src/azure-cli/azure/cli/command_modules/profile/custom.py

```diff
@@ -59,27 +59,30 @@ def show_subscription(cmd, subscription=None, show_auth_for_sdk=None):
     print(json.dumps(profile.get_sp_auth_info(subscription), indent=2))
 
 
-def get_access_token(cmd, subscription=None, resource=None, resource_type=None):
-    '''
+def get_access_token(cmd, subscription=None, resource=None, resource_type=None, tenant=None):
+    """
     get AAD token to access to a specified resource
     :param resource: Azure resource endpoints. Default to Azure Resource Manager
     :param resource-type: Name of Azure resource endpoints. Can be used instead of resource.
     Use 'az cloud show' command for other Azure resources
-    '''
+    """
     if resource is None and resource_type is not None:
         endpoints_attr_name = cloud_resource_type_mappings[resource_type]
         resource = getattr(cmd.cli_ctx.cloud.endpoints, endpoints_attr_name)
     else:
         resource = (resource or cmd.cli_ctx.cloud.endpoints.active_directory_resource_id)
     profile = Profile(cli_ctx=cmd.cli_ctx)
-    creds, subscription, tenant = profile.get_raw_token(subscription=subscription, resource=resource)
-    return {
+    creds, subscription, tenant = profile.get_raw_token(subscription=subscription, resource=resource, tenant=tenant)
+
+    result = {
         'tokenType': creds[0],
         'accessToken': creds[1],
         'expiresOn': creds[2].get('expiresOn', 'N/A'),
-        'subscription': subscription,
         'tenant': tenant
     }
+    if subscription:
+        result['subscription'] = subscription
```

Comment:
Seems we changed previous behavior here. `subscription` property value is N/A before, after this change, the property will be removed. Is this a break change to some customers ?

Reply:
Nope. Previously there is no way for the user to get a token for a tenant, so this is a new feature rather than a breaking change.

Reply:
Nope. Previously there is no way for the user to get a token for a tenant, so this is a new feature rather than a breaking change. 

The previous behavior for `az account get-access-token --subscription` is not affected. This can be verified by the test cases.